src/components/steps/StepSearchParty.tsx

```diff
@@ -101,6 +101,57 @@ export function StepSearchParty({
     }
   };
 
+  const subunitTypeByQuery = new URLSearchParams(window.location.search).get('subunitType') ?? '';
+  const subunitCodeByQuery = new URLSearchParams(window.location.search).get('subunitCode') ?? '';
+  const prodPn = product?.id === 'prod-pn';
+
+  const handleSearchByAooCode = async (query: string) => {
+    const searchResponse = await fetchWithLogs(
+      { endpoint: 'ONBOARDING_GET_AOO_CODE_INFO', endpointParams: { codiceUniAoo: query } },
+      {
+        method: 'GET',
+        params: { ...(prodPn && { categories: 'L6,L4,L45', origin: 'IPA' }) },
+      },
+      () => setRequiredLogin(true)
+    );
+
+    const outcome = getFetchOutcome(searchResponse);
+
+    if (outcome === 'success') {
+      setAooResult((searchResponse as AxiosResponse).data);
+      setAooResultHistory((searchResponse as AxiosResponse).data);
+    } else if ((searchResponse as AxiosError).response?.status === 404) {
+      setAooResult(undefined);
+    }
+  };
+  const handleSearchByUoCode = async (query: string) => {
```

Comment:
You can merge the handleSearchByAooCode and handleSearchByUoCode in an unique method that take the subunitTypeByQuery as second param, with this for use conditionally an endpoint instead of another